WebDriverAgentLib/Categories/XCUIDevice+FBHelpers.m

```diff
@@ -40,6 +63,48 @@ - (BOOL)fb_goToHomescreenWithError:(NSError **)error
   return YES;
 }
 
+- (BOOL)fb_lockScreen:(NSError **)error
+{
+  if (locked) {
+    return TRUE;
+  }
+  SEL mSelector = NSSelectorFromString(@"pressLockButton");
+  [self methodSignatureForSelector:mSelector];
+  NSMethodSignature *mSignature = [self methodSignatureForSelector:mSelector];
+  NSInvocation *invocation = [NSInvocation invocationWithMethodSignature:mSignature];
+  [invocation setTarget:self];
+  [invocation setSelector:mSelector];
+  [invocation invoke];
+  return [[[[FBRunLoopSpinner new]
+            timeout:10.]
+           timeoutErrorMessage:@"Timeout waiting until screen getting locked"]
+          spinUntilTrue:^BOOL{
+            return locked;
+          } error:error];
+}
+
+- (BOOL)fb_isScreenLocked
+{
+  return locked;
+}
+
+- (BOOL)fb_unlockScreen:(NSError **)error
```

Comment:
this will, most likely, fail for iOS < 10, since it requires single swipe in order to unlock the screen